src/reducerMaps/restaurants.js

```diff
@@ -70,12 +70,15 @@ export default new Map([
     })
   ],
   [ActionTypes.POST_RESTAURANT, isFetching],
-  [ActionTypes.RESTAURANT_POSTED, (state, action) =>
-    update(state, {
+  [ActionTypes.RESTAURANT_POSTED, (state, action) => {
+    const updates = {
       isFetching: {
         $set: false
-      },
-      items: {
+      }
+    };
+
+    if (!(action.restaurant.id in state.items.entities.restaurants)) {
```

Comment:
It's possible in rare cases for `state.items.entities.restaurants` to not exist (see line 86). Make sure you check here as well. 

Also, don't use `in`. Check if `indexOf` is `-1` instead.